feat(docs): Fixes from audit (#7640)
AztecProtocol/aztec-packages: docs/docs/guides/smart_contracts/writing_contracts/how_to_prove_history.md

> The Aztec Protocol uses an append-only Merkle tree to store hashes of the headers of all previous blocks in the chain as its leaves. This is known as an archive tree. You can learn more about how it works in the [concepts section](../../../aztec/concepts/storage/trees/index.md#archive-tree).

Reviewer: ```suggestion
The Aztec Protocol uses an append-only Merkle tree to store hashes of the headers of all previous blocks in the chain as its leaves. This is known as the Archive tree. You can learn more about how it works in the [concepts section](../../../aztec/concepts/storage/trees/index.md#archive-tree).
```

This page is a quick introductory guide to creating historical proofs proofs from the archive tree.

For a reference, go [here](https://docs.aztec.network/reference/smart_contract_reference/aztec-nr/aztec/history/contract_inclusion).

> # Inclusion and non-inclusion proofs

Reviewer: ```suggestion
## Inclusion and non-inclusion proofs
```
because otherwise it looks weird with text above the top-level heading 
![image](https://github.com/user-attachments/assets/d3fc0225-67be-41d6-8e00-9c0b4758f36f)

> Inclusion and non-inclusion proofs are available in the header of a block. You can prove any of the following at a given block height before the current height:

Reviewer: ```suggestion
Inclusion and non-inclusion proofs refer to proving the inclusion (or absence) of a specific piece of information within a specific Aztec block with a block header. You can prove any of the following at a given block height before the current height:
```

Reviewer: The proofs aren't technically available in the header, but the headers are used to create a proof.
